pipeline/matcher/src/main/scala/weco/pipeline/matcher/matcher/WorkMatcher.scala

```diff
@@ -35,7 +35,7 @@ class WorkMatcher(
         beforeNodes <- workGraphStore.findAffectedWorks(work)
         afterNodes = WorkGraphUpdater.update(work, beforeNodes)
 
-        updatedNodes = afterNodes -- beforeNodes
+        updatedNodes = afterNodes.flatten -- beforeNodes
```

Comment:
Are we `flatten`ing `afterNodes` 3 times? Can we just return it in flattened form?